### DISCUSSION TITLE
Handling changlog window on startup

### DISCUSSION THREAD

**Discussion by mmattel** [zwave-js/zwave-js-ui#3363] [General]
**Is your feature request related to a problem? Please describe.**
Handling changlog window on startup

**Describe the solution you'd like**
Move the "dont show + ok block" from the bottom to the top, else you always need to fully scroll down to close it.

**Describe alternatives you've considered**
Scrolling down to the bottom and over time the changelog list will be big which then becomes annoying.

**Additional context**
I like the changlog being presented but scrolling down just to close it where I am interested in the latest changes is really not that good.

**Comment by robertsLando** [comment]
@mmattel The don't show disables it forever, not just for the actual release. Once you press ok you will not see that changelog anymore until next release

**Comment by mmattel** [comment]
I know, I suggest to have that block on top

**Comment by robertsLando** [comment]
Reason it is on bottom is that reading changelogs is highly suggested so you should scroll to bottom in order to read it all, like when you have to accept terms in a website it force you to scroll the entire page in order to go on. I would like to know others opinion on this as I'm open  to change this behaviour but I actually did it on purpose for the reasons described above

**Comment by AlCalzone** [comment]
> where I am interested in the latest changes

I don't understand the request then. The changelog should show only the changes that were made since you read them last.

**Comment by mmattel** [comment]
> The changelog should show only the changes that were made since you read them last.

If the changelog would only show the changes of the particular release, I would be fine with the current state. But as long it shows all changes, imagine +20 releasen in the future, scrolling the list to the bottom for options becomes a nightmare.

**Comment by robertsLando** [comment]
It has a limit of 10 releases btw I can also reduce this limit, that said in the worse case the scroll has to be done just one time every time you do a release, if you have 10 releases means you don't update it so often

Also you can disable it and since someone keeps it enabled means he want to see it and scrolling would not be a problem.

This is just my opinion but I may be wrong 🙌🏻

**Comment by mmattel** [comment]
> Also you can disable it and since someone keeps it enabled means he want to see it and scrolling would not be a problem.

This assupmtion is imho not correct. One wants to look for the changes of the current release and not for all in the list. Disabling would mean not showing the list at all - even not for the current release. Limiting the amount of changelogs shown could be a alternative solution, but the number should be quite low, I would say 2. The current and the former.

**Comment by robertsLando** [comment]
@mmattel In the changelog you see all changes made from the version you was to the one you just installed so IMO say "I want to see only latest changes" makes no sense. I mean for example if you update to a patch release with just a minor bug fix you would miss all the intermediate important changes 🤷🏼‍♂️  

The reason to see multiple releases is to show you every single feature/patch made since your last update and that's the main purpose of a changelog

**Comment by AlCalzone** [comment]
> One wants to look for the changes of the current release and not for all in the list

So, if you were to update from v8.26.0 to v9.1.2, you would only want to see this
> add setting to enable new version notification

and not the breaking changes that happened in v9.0.0 and the explanation what you might have to do to get things fixed again?
> Minimum NodeJS version is now 18
> If ZUI doesn’t work as expected after updating please check this issue and follow the steps described here: https://github.com/zwave-js/node-zwave-js/issues/6341
> ... etc

That seems wrong IMO.

**Comment by mmattel** [comment]
I think that all post contain valuable info to improve the current behaviour.
My initial attempt was just to move the selecors what to do next from the bottom to the top.
This would keep the list of changes, as log how ever defined, and makes it easier for the reader to do the next step.
